[41437] Use signalling on project edit field

oliverguenther opened the pull request:
https://community.openproject.org/wp/41437

Commit:
Add project-edit-field that uses op-autocompleter

oliverguenther committed:
Replace for_type param with regular filter

oliverguenther committed:
Fix unrelated spec now fixed in behavior due to the select helper change

The spec did not apply the new type previously, so would not override
the description. This caused the expects to remain valid, but was
actually incorrect behavior.

oliverguenther committed:
Ensure we're not focussing the project field on inline create

cbliard commented:
There is a feature test being flaky since commit 3f19bbeb8eb9b2 (the first one).

```
Failed examples:

rspec ./spec/features/custom_fields/multi_user_custom_field_spec.rb:61 # multi select custom values with mixed users, group, and placeholdders should be shown and allowed to be updated
```

it fails since [this modification](https://github.com/opf/openproject/pull/10432/commits/3f19bbeb8eb9b237b2a224a8e0f8f1df5693648e#diff-91d2fb7fb1b71e21f4f6c238531057d443a94195994ae4de00f6512d88606b15L136-L138): `page.find('.ng-dropdown-panel .ng-option', visible: :all, text: content).click` being replaced by `select_autocomplete field_container, query: content, results_selector: 'body'`.

The code sends key "Da Real" to the input, but the input value is "DaReal". As the input text does not match any entries, nothing is displayed and the right select entry cannot be clicked on.

I tried investigating, it looks like we face this issue: https://github.com/ng-select/ng-select/issues/1869 . The list of items is empty while it loads, and so spaces cannot be input in the field (detailed explanation in linked issue).

If adding a delay, the values are loaded into the component and when sending keys the input value is correct and the test passes.

cbliard committed:
fix flickering test

because of ng-select/ng-select#1869 we can't use a string with spaces in ng-select input value while the list of items is empty. The spaces would be ignored and matching an entry would then fail. Instead wait for ng-dropdown-panel to appear which means items are present, the bug is inactive, and the spaces can be properly sent to the input.

flickering test was spec/features/custom_fields/multi_user_custom_field_spec.rb:61

cbliard commented:
apart from that, I tested the feature and it now displays lots of projects (my local install was limited to 20 entries when I tried against `dev`).

Also it displays all projects, and not just subprojects of the current project. Is it the correct behavior?

cbliard committed:
fix tests broken in 65695f1

- Reintroduce sleep to make test pass. I'd like to get rid of it but
  can't understand why it's needed.
- refine results_selector only if it is 'body'.

cbliard committed:
fix tests by not waiting for dropdown before typing for global search

oliverguenther commented:
The remaining test failing is that temporary one from dev
